### ISSUE TITLE
Threshold Period Applied Times causing DateTimePareException

### ISSUE THREAD

**Issue by zmoore-usgs** [AquaticInformatics/aquarius-sdk-java#35]
Hello,

I'm currently testing out the Aquarius SDK as a replacement for our current home-built data-retrieval system for our Aquarius instance and I've hit a bit of a snag. It seems that the times coming back in the `AppliedTime` field of `Threshold` `Periods` (Full JSON Path: TimeSeriesDescriptions[i].Thresholds.Periods.AppliedTime) of a `Publish.TimeSeriesDescriptionServiceRequest` are not being properly deserialized. The time formats we're getting back for this field seem to not have any trailing 0s in order to get us to the required number of digits in the sub-second portion of the date. I'm not sure if this is a setting we need to change in our Aquarius system, a bug in the way Aquarius is outputting dates, or if perhaps the deserializer here should be made a bit more flexible to be able to handle times that don't have trailing 0s?

Example of the time format differences:

```
Period Start Time that does properly deserialize: 2001-10-01T00:00:00.0000000-05:00
Period Applied Time that does not properly deserialize: 2017-02-02T21:01:15.3558Z
```

Any help with this would be greatly appreciated! In the meantime I'm going to test out forking this project and modifying the Instant serializer to support missing trailing 0s. Not sure if that is the ideal solution or not though, so I'll wait to hear from you all first before I treat this issue as resolved.

**Comment by zmoore-usgs** [comment]
I created a fork and did a little test to see if I could fix the issue and my changes seem to work, though I'm not sure if they're the best way to go about fixing this. I can create a PR here if I'm on the right track though (again not sure if this is an issue with the SDK deserialization or an issue with the actual dates being returned by our Aquarius instance - if it's the latter then perhaps this change shouldn't be made to the SDK).

https://github.com/zmoore-usgs/aquarius-sdk-java/compare/master...zmoore-usgs:bug/pad-times

**Comment by schmidt4brains** [comment]
Hmm. This is just a bug.

I was expecting partial fractional seconds to deserialize without drama, anything from 0 to 7 fractional second digits). But clearly that case isn't covered by tests.

I'll bet it is an easy fix. I'll take a look.

**Comment by schmidt4brains** [comment]
Oh, I see the problem.

Some of the properties being serialized on the server are .NET `DateTime` types, like [Approval.DateAppliedUtc](https://github.com/AquaticInformatics/aquarius-sdk-net/blob/develop/src/Aquarius.Client/TimeSeries/Client/ServiceModels/Publish.cs#L144)

And other properties like [PeriodOfApplicability.StartTime](https://github.com/AquaticInformatics/aquarius-sdk-net/blob/develop/src/Aquarius.Client/TimeSeries/Client/ServiceModels/Publish.cs#L1162) are .NET `DateTimeOffset` types.

And yet others still like [LocationDatumPeriod.AppliedTimeUtc](https://github.com/AquaticInformatics/aquarius-sdk-net/blob/develop/src/Aquarius.Client/TimeSeries/Client/ServiceModels/Publish.cs#L745) are NodaTime `Instant` types.

All 3 types are mapped to a `java.time.Instant` by the Java SDK.

`Instant` and `DateTimeOffset` will be serialized identically on the server, but `DateTime` ISO 8601 values (any timestamp you see ending with a `Z`) can serialize with an optional number of fractional seconds.

I'll need to tweak the SDK to account for that.

**Comment by schmidt4brains** [comment]
The fix for this has been deployed in v17.4.3 in the Maven repo

**Comment by zmoore-usgs** [comment]
Thanks for the quick fix!
